src/lib/swapper/types.ts

```diff
@@ -284,7 +285,7 @@ export type SwapperApi = {
   checkTradeStatus: (
     input: CheckTradeStatusInput,
   ) => Promise<{ status: TxStatus; buyTxHash: string | undefined; message: string | undefined }>
-  getTradeQuote: (input: GetTradeQuoteInput) => Promise<TradeQuoteResult>
+  getTradeQuote: (input: GetTradeQuoteInput, assetsById: AssetsById) => Promise<TradeQuoteResult>
```

Comment:
architecturally this goes a little backwards with what we were trying to achieve with keeping `getTradeQuote` as pure as possible

in the future, on the API side, the API will have access to the assets, via the same kind of mechanism of `new AssetService()`

if we merge this, we'll just have to remember to revert this out at a future date

how much perf do we gain from avoiding this?

Reply:
Discussed this one out, we want to be as lazy as we can here, we can always revert this PR out - added somme commentary https://github.com/shapeshift/web/pull/5519/commits/8af68efa34539fb1f3f877450602335271ead3e3